compiler/packages/babel-plugin-react-compiler/src/Inference/InferEffectDependencies.ts

```diff
@@ -169,8 +170,17 @@ export function inferEffectDependencies(fn: HIRFunction): void {
       ) {
         const callee =
           value.kind === 'CallExpression' ? value.callee : value.property;
+
+        const autodepsArgIndex = value.args.findIndex(
+          arg =>
+            arg.kind === 'Identifier' &&
+            arg.identifier.type.kind === 'Object' &&
+            arg.identifier.type.shapeId === BuiltInAutodepsId,
+        );
         if (
-          value.args.length === autodepFnLoads.get(callee.identifier.id) &&
+          value.args.length > 1 &&
+          autodepsArgIndex > 0 &&
```

Comment:
In the next diff I change the config to specify the index for autodeps instead of number of required args